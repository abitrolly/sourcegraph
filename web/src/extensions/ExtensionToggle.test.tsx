import React from 'react'
import { ConfiguredRegistryExtension } from '../../../shared/src/extensions/extension'
import { PlatformContext } from '../../../shared/src/platform/context'
import { ConfiguredSubjectOrError, SettingsSubject } from '../../../shared/src/settings/settings'
import { ExtensionToggle } from './ExtensionToggle'
import { mount } from 'enzyme'

describe('ExtensionToggle', () => {
    const NOOP_PLATFORM_CONTEXT: PlatformContext = {} as any
    const SUBJECT: ConfiguredSubjectOrError = {
        lastID: null,
        settings: {},
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        subject: { __typename: 'User', id: 'u', viewerCanAdminister: true } as SettingsSubject,
    }
    const EXTENSION: Pick<ConfiguredRegistryExtension, 'id'> = {
        id: 'x/y',
    }

    test('extension not present in settings', () => {
        expect(
            mount(
                <ExtensionToggle
                    extension={EXTENSION}
                    settingsCascade={{ final: { extensions: {} }, subjects: [SUBJECT] }}
                    platformContext={NOOP_PLATFORM_CONTEXT}
                />
            )
        ).toMatchSnapshot()
    })

    test('extension enabled in settings', () => {
        expect(
            mount(
                <ExtensionToggle
                    extension={EXTENSION}
                    settingsCascade={{ final: { extensions: { 'x/y': true } }, subjects: [SUBJECT] }}
                    platformContext={NOOP_PLATFORM_CONTEXT}
                />
            )
        ).toMatchSnapshot()
    })

    test('extension disabled in settings', () => {
        expect(
            mount(
                <ExtensionToggle
                    extension={EXTENSION}
                    settingsCascade={{ final: { extensions: { 'x/y': false } }, subjects: [SUBJECT] }}
                    platformContext={NOOP_PLATFORM_CONTEXT}
                />
            )
        ).toMatchSnapshot()
    })
})
