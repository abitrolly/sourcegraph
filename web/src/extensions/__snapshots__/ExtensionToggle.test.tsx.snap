// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtensionToggle extension disabled in settings 1`] = `
<ExtensionToggle
  extension={
    Object {
      "id": "x/y",
    }
  }
  platformContext={Object {}}
  settingsCascade={
    Object {
      "final": Object {
        "extensions": Object {
          "x/y": false,
        },
      },
      "subjects": Array [
        Object {
          "lastID": null,
          "settings": Object {},
          "subject": Object {
            "__typename": "User",
            "id": "u",
            "viewerCanAdminister": true,
          },
        },
      ],
    }
  }
>
  <Toggle
    onToggle={[Function]}
    title="Click to enable"
    value={false}
  >
    <button
      className="toggle  "
      onClick={[Function]}
      title="Click to enable"
      type="button"
      value={0}
    >
      <span
        className="toggle__bar  "
      />
      <span
        className="toggle__knob  "
      />
    </button>
  </Toggle>
</ExtensionToggle>
`;

exports[`ExtensionToggle extension enabled in settings 1`] = `
<ExtensionToggle
  extension={
    Object {
      "id": "x/y",
    }
  }
  platformContext={Object {}}
  settingsCascade={
    Object {
      "final": Object {
        "extensions": Object {
          "x/y": true,
        },
      },
      "subjects": Array [
        Object {
          "lastID": null,
          "settings": Object {},
          "subject": Object {
            "__typename": "User",
            "id": "u",
            "viewerCanAdminister": true,
          },
        },
      ],
    }
  }
>
  <Toggle
    onToggle={[Function]}
    title="Click to enable"
    value={true}
  >
    <button
      className="toggle  "
      onClick={[Function]}
      title="Click to enable"
      type="button"
      value={1}
    >
      <span
        className="toggle__bar toggle__bar--active "
      />
      <span
        className="toggle__knob toggle__knob--active "
      />
    </button>
  </Toggle>
</ExtensionToggle>
`;

exports[`ExtensionToggle extension not present in settings 1`] = `
<ExtensionToggle
  extension={
    Object {
      "id": "x/y",
    }
  }
  platformContext={Object {}}
  settingsCascade={
    Object {
      "final": Object {
        "extensions": Object {},
      },
      "subjects": Array [
        Object {
          "lastID": null,
          "settings": Object {},
          "subject": Object {
            "__typename": "User",
            "id": "u",
            "viewerCanAdminister": true,
          },
        },
      ],
    }
  }
>
  <Toggle
    onToggle={[Function]}
    title="Click to enable"
    value={false}
  >
    <button
      className="toggle  "
      onClick={[Function]}
      title="Click to enable"
      type="button"
      value={0}
    >
      <span
        className="toggle__bar  "
      />
      <span
        className="toggle__knob  "
      />
    </button>
  </Toggle>
</ExtensionToggle>
`;
